packages/status-js/src/request-client/request-client.ts

```diff
@@ -307,12 +307,21 @@ class RequestClient {
       // eslint-disable-next-line no-empty
     }
 
-    const decodedMetadata =
-      ApplicationMetadataMessage.fromBinary(messageToDecode)
+    let decodedMetadata
+    try {
+      decodedMetadata = ApplicationMetadataMessage.fromBinary(messageToDecode)
+    } catch {
+      return
+    }
+
     if (!decodedMetadata.payload) {
       return
     }
 
+    if (!decodedMetadata.signature.length) {
+      return
+    }
+
```

Comment:
Overall, I think the code is a bit verbose and makes it harder to read and maintain. Maybe joining these conditions will help a bit.

```typescript
  if (!decodedMetadata || !decodedMetadata.payload || !decodedMetadata.signature.length) {
    return;
  }
  ```

Also i see you're not handling the error. Wouldn't you use it somehow in this case?

Reply:
In the current context, it's an invalid message that's worth skipping, I believe. I do think it's worth monitoring invalid messages, their size and rate but on anohter level, I'd say.